// ng4-file-upload/src/fileuploader.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
var FileUploaderComponent = /** @class */ (function () {
    function FileUploaderComponent() {
        this.onSelection = new EventEmitter();
        this.DisplayedText = "";
    }
    FileUploaderComponent.prototype.fileChange = function (event) {
        var fileList = event.target.files;
        var filetypeToCompare = this.fileType.replace('*', '');
        var hasFile = fileList && fileList.length > 0;
        if (hasFile) {
            if (filetypeToCompare != "application/" && fileList[0].type.indexOf(filetypeToCompare) == -1) {
                fileList = null;
                this.DisplayedText = "";
            }
            else {
                if (this.maxSizeInKb)
                    for (var i = 0; i < fileList.length; i++) {
                        if (fileList[i].size > this.maxSizeInKb * 1024) {
                            alert("File size is more than " + this.maxSizeInKb + " Kb");
                            fileList = null;
                            this.DisplayedText = "";
                            return false;
                        }
                    }
                var multipleFile = fileList.length > 1;
                if (multipleFile) {
                    this.DisplayedText = fileList.length + ' file(s) has been selected';
                }
                else {
                    var file = fileList[0];
                    this.DisplayedText = file.name;
                }
            }
            this.onSelection.emit(fileList);
        }
    };
    FileUploaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'file-uploader',
                    template: "<div class=\"btn-toolbar mb-3\" role=\"toolbar\">\n    <div class=\"input-group\">\n        <div class=\"input-group-prepend\">\n            <label class=\"btn btn-primary\">\n                <span>Browse</span>...\n                <input type=\"file\" accept=\"{{fileType}}\" style=\"display: none;\" (change)=\"fileChange($event)\" [attr.multiple]=\"allowMultiple\">\n            </label>\n        </div>\n        <input type=\"text\" class=\"form-control\" [value]=\"DisplayedText\" disabled=\"\" [class.ng-invalid-right]=\"required && !DisplayedText\"\n            [class.ng-valid-right]=\"required && DisplayedText\">\n    </div>\n</div>",
                    styles: ["\n        label.btn{  margin: 0;}"]
                },] },
    ];
    /** @nocollapse */
    FileUploaderComponent.ctorParameters = function () { return []; };
    FileUploaderComponent.propDecorators = {
        "allowMultiple": [{ type: Input },],
        "fileType": [{ type: Input },],
        "required": [{ type: Input },],
        "maxSizeInKb": [{ type: Input },],
        "onSelection": [{ type: Output },],
    };
    return FileUploaderComponent;
}());
export { FileUploaderComponent };
;
