// ng4-file-upload/src/app/file-uploader/src/fileuploader.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var FileUploaderComponent = /** @class */ (function () {
    function FileUploaderComponent() {
        this.onSelection = new core_1.EventEmitter();
        this.DisplayedText = "";
    }
    FileUploaderComponent.prototype.fileChange = function (event) {
        var fileList = event.target.files;
        var filetypeToCompare = this.fileType.replace('*', '');
        var hasFile = fileList && fileList.length > 0;
        if (hasFile) {
            if (filetypeToCompare != "application/" && fileList[0].type.indexOf(filetypeToCompare) == -1) {
                fileList = null;
                this.DisplayedText = "";
            }
            else {
                if (this.maxSizeInKb)
                    for (var i = 0; i < fileList.length; i++) {
                        if (fileList[i].size > this.maxSizeInKb * 1024) {
                            alert("File size is more than " + this.maxSizeInKb + " Kb");
                            fileList = null;
                            this.DisplayedText = "";
                            return false;
                        }
                    }
                var multipleFile = fileList.length > 1;
                if (multipleFile) {
                    this.DisplayedText = fileList.length + ' file(s) has been selected';
                }
                else {
                    var file = fileList[0];
                    this.DisplayedText = file.name;
                }
            }
            this.onSelection.emit(fileList);
        }
    };
    FileUploaderComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'file-uploader',
                    templateUrl: 'fileuploader.html',
                    styleUrls: [
                        '../../../../node_modules/bootstrap/dist/css/bootstrap.min.css',
                        './fileuploader.component.css'
                    ]
                },] },
    ];
    /** @nocollapse */
    FileUploaderComponent.ctorParameters = function () { return []; };
    FileUploaderComponent.propDecorators = {
        "allowMultiple": [{ type: core_1.Input },],
        "fileType": [{ type: core_1.Input },],
        "required": [{ type: core_1.Input },],
        "maxSizeInKb": [{ type: core_1.Input },],
        "onSelection": [{ type: core_1.Output },],
    };
    return FileUploaderComponent;
}());
exports.FileUploaderComponent = FileUploaderComponent;
;
